send/benchmark/harness_case.go

```
@@ -0,0 +1,139 @@
+package send
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"time"
+)
+
+type caseDefinition struct {
+	name               string
+	bench              benchCase
+	count              int
+	numOps             int
+	size               int
+	requiredIterations int
+	runtime            time.Duration
+
+	cumulativeRuntime time.Duration
+	elapsed           time.Duration
+	startAt           time.Time
+	isRunning         bool
+}
+
+// TimerManager is a subset of the testing.B tool, used to manage
+// setup code.
+type TimerManager interface {
+	ResetTimer()
+	StartTimer()
+	StopTimer()
+}
+
+func (c *caseDefinition) ResetTimer() {
+	c.startAt = time.Now()
+	c.elapsed = 0
+	c.isRunning = true
+}
+
+func (c *caseDefinition) StartTimer() {
+	c.startAt = time.Now()
+	c.isRunning = true
+}
+
+func (c *caseDefinition) StopTimer() {
+	if !c.isRunning {
+		return
```

This could cause the code to silently ignore errors. If no caller should ever call this function in this case, then it might be better to `panic("StopTimer called on stopped timer")`

---

This is also from the original Go driver code. I changed it to panic.